parquet/src/test/java/org/apache/iceberg/parquet/TestBloomRowGroupFilter.java

```diff
@@ -1189,4 +1192,40 @@ public void testTransformFilter() {
         .as("Should read: filter contains non-reference evaluate as True")
         .isTrue();
   }
+
+  @Test
+  public void testParquetFindsResidual() {
+    // `string` col has no dict, this should be eliminated by bloom filter
+    Expression bloom = equal("string", "BINARY测试_301");
+    // should be eliminated by dictionary filter
+    Expression dict = equal("no_stats", "a");
+    // should be eliminated by bloom filter
+    Expression metric = greaterThan("id", INT_MAX_VALUE);
+    Expression expr = or(bloom, or(dict, metric));
+
+    Expression expected = Binder.bind(SCHEMA.asStruct(), Expressions.or(bloom, dict), true);
+    ParquetMetricsRowGroupFilter metricFilter = new ParquetMetricsRowGroupFilter(SCHEMA, expr);
+    Expression metricResidual = metricFilter.residualFor(parquetSchema, rowGroupMetadata);
+    assertThat(expected.isEquivalentTo(metricResidual))
+        .as("Expected residual: %s, actual residual: %s", expected, metricResidual);
+
+    expected = Binder.bind(SCHEMA.asStruct(), bloom, true);
+    ParquetDictionaryRowGroupFilter dictFilter =
+        new ParquetDictionaryRowGroupFilter(SCHEMA, metricResidual);
+    Expression dictResidual =
+        dictFilter.residualFor(
+            parquetSchema, rowGroupMetadata, reader.getDictionaryReader(rowGroupMetadata));
+
+    assertThat(expected.isEquivalentTo(dictResidual))
+        .as("Expected residual: %s, actual residual: %s", expected, dictResidual);
+
+    expected = Expressions.alwaysFalse();
+    ParquetBloomRowGroupFilter bloomFilter = new ParquetBloomRowGroupFilter(SCHEMA, dictResidual);
+    Expression bloomResidual =
+        bloomFilter.residualFor(
+            parquetSchema, rowGroupMetadata, reader.getBloomFilterDataReader(rowGroupMetadata));
+
```

Comment:
It feels a bit odd to test the combined filter residual logic in this test method in `TestBloomRowGroupFilter`. See my comment above on a separate test class which encapsulates the logic, which should also make it easier for testing since we can then move this to a separate test class.

As far as testing goes, here are the cases I think:

1.) We know the existing tests should cover the cases where an individual filter is always true or always false.
2.) So following 1, then we'd want the following tests for the combined tests:
      a.) Where the metrics filter has a residual that's always true/false
      b.) Where the metrics filter has a residual that's not true/false and the dictionary filter has one that is true/false.
      c.) Where the metrics filter has a residual that's not true/false, the dictionary filter does not have one that is true/false, and the bloom filter returns a residual that's not true/false. 
      d.) Same as c but the bloom filter does return true/false

Reply:
Hi @amogh-jahagirdar, thanks for the comments.

I'm sorry for the delay, comments have been addressed, please take a look when you have time.